packages/cli/commands/core.py

```diff
@@ -61,14 +71,22 @@ def _start_pyra_core() -> None:
     existing_pid = _process_is_running()
     if existing_pid is not None:
         _print_red(f"Background process already exists with PID {existing_pid}")
-    else:
-        p = subprocess.Popen(
-            [_INTERPRETER_PATH, _CORE_SCRIPT_PATH],
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-        )
-        utils.Logger.log_activity_event("start-core")
-        _print_green(f"Started background process with PID {p.pid}")
+        return
+
+    try:
+        _run_pyra_core_lock.acquire()
+        _run_pyra_core_lock.release()
+    except filelock.Timeout:
```

Comment:
Is this the best way to do this?

Reply:
I think so.

`run_pyra_core.py` will acquire that lock before the mainloop starts and frees it when the program finishes or crashes: https://github.com/tum-esm/pyra/blob/52eb662f0bfb33d224daf9cf7cfe6cce192da732/run_pyra_core.py#L14-L22

When the `pyra-cli start` command cannot acquire the Lock, some Pyra Core process must be running.

The other option would be to cycle through all running processes (like the `is-running` command), but I think that is more work to implement because there are different cases.